ehr/src/org/labkey/ehr/EHRServiceImpl.java

```diff
@@ -1011,4 +1027,25 @@ public void appendSNOMEDCols(AbstractTableInfo ti, String displayColumnName, Str
             ti.addColumn(simpleRawCol);
         }
     }
+
+    @Override
+    public void registerDataEntryFormAccess(Map<Module, Map<String, Set<Class<? extends DataEntryForm>>>> permittedForms)
```

Comment:
This overall pattern looks good, but I propose that we create a DataEntryFormFilter interface with a single method, isAvailable(DataEntryForm, Container, User). TNPRC can implement that interface with its group/form name logic